rust-symcrypt/src/rsa/mod.rs

```diff
@@ -0,0 +1,716 @@
+            // When only getting the public key, ppPrimes, pcbPrimes and nPrimes can be NULL, NULL and 0.
+            match symcrypt_sys::SymCryptRsakeyGetValue(
+                self.inner.0,
+                modulus_buffer.as_mut_ptr(),
+                modulus_buffer.len() as symcrypt_sys::SIZE_T,
+                pub_exp.as_mut_ptr(),
+                1 as symcrypt_sys::UINT32,
+                ptr::null_mut(),
+                ptr::null_mut(),
+                0,
+                NumberFormat::MSB.to_symcrypt_format(),
+                self.key_usage.to_symcrypt_flag(),
+            ) {
+                symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR => Ok(RsaPublicKeyExportBlob {
+                    modulus: modulus_buffer,
+                    pub_exp: store_msb_first_u64(pub_exp[0], self.get_size_of_public_exponent())?,
+                }),
+                err => Err(err.into()),
+            }
+        }
+    }
+
+    /// `get_size_of_modulus()` returns a `u32` representing the (tight) size in bytes of a byte array big enough to store
+    /// the modulus of the key.
+    pub fn get_size_of_modulus(&self) -> u32 {
+        unsafe {
+            // SAFETY: FFI calls
+            symcrypt_sys::SymCryptRsakeySizeofModulus(self.inner.0)
+        }
+    }
+
+    /// `get_size_of_public_exponent()` returns a `u32` representing the (tight) size in bytes of a byte array big enough to store
+    /// the public exponent of the key.
+    pub fn get_size_of_public_exponent(&self) -> u32 {
+        unsafe {
+            // SAFETY: FFI calls
+            // Only one public exponent is supported, so the only valid index is 0.
+            symcrypt_sys::SymCryptRsakeySizeofPublicExponent(self.inner.0, 0)
+        }
+    }
+
+    /// `get_key_usage` returns the intended usage for the RSA key pair.
+    pub fn get_key_usage(&self) -> RsaKeyUsage {
+        self.key_usage
+    }
+
+    // `inner` gives crate access to the inner symcrypt Rsa Key struct.
+    pub(crate) fn inner(&self) -> symcrypt_sys::PSYMCRYPT_RSAKEY {
+        self.inner.0
+    }
+}
+
+impl RsaPublicKey {
+    /// `set_public_key()` sets only the public key information onto the [`RsaPublicKey`].
+    ///
+    /// `modulus_buffer` takes in a `&[u8]` reference to a byte array that contains the modulus of the Rsa key.
+    ///
+    /// `pub_exp` takes in a `&[u8]` that is an array of bytes representing the public exponent.
+    ///
+    /// `key_usage` takes in a [`RsaKeyUsage`] and will indicate if this key will be used for [`RsaKeyUsage::Sign`], or [`RsaKeyUsage::Encrypt`], or [`RsaKeyUsage::SignAndEncrypt`]
+    pub fn set_public_key(
+        modulus_buffer: &[u8],
+        pub_exp: &[u8],
+        key_usage: RsaKeyUsage,
+    ) -> Result<Self, SymCryptError> {
+        let n_bits_mod = (modulus_buffer.len() as u32) * 8; // Convert the size from bytes to bits
+        let rsa_key = allocate_rsa(0, n_bits_mod)?;
+        let u64_pub_exp = load_msb_first_u64(pub_exp)?;
```

Comment:
If you require the caller to pass in a pub_exp that is u64 value, then, you will need to provide a public function for and document.